datafusion/core/src/dataframe.rs

```diff
@@ -1007,4 +1047,37 @@ mod tests {
         .await?;
         Ok(())
     }
+
+    #[tokio::test]
+    async fn with_column() -> Result<()> {
+        let df = test_table().await?.select_columns(&["c1", "c2", "c3"])?;
+        let ctx = SessionContext::new();
+        let df_impl = Arc::new(DataFrame::new(ctx.state.clone(), &df.plan.clone()));
+
+        // check that we correctly read from the table
+        let df_results = &df_impl
+            .filter(col("c2").eq(lit(3)))?
+            .filter(col("c1").eq(lit("a")))?
```

Comment:
`Expr` has `and` and `or` implemented so you should be able to write it like this?

```suggestion
            .filter(col("c2").eq(lit(3)).and(col("c1").eq(lit("a"))))?
```